ui/app/contexts/metametrics.js

```diff
@@ -55,45 +52,65 @@ export function MetaMetricsProvider ({ children }) {
     return unlisten
   }, [history])
 
+  /**
+   * track a metametrics event
+   *
+   * @param {import('../../../shared/modules/metametrics').MetaMetricsEventPayload} - payload for event
+   * @returns undefined
+   */
+  const trackEvent = useMemo(() => {
+    const referrer = confirmTransactionOrigin ? { url: confirmTransactionOrigin } : undefined
+    const page = {
+      path: currentPath,
+    }
+    return getTrackMetaMetricsEvent(global.platform.getVersion(), () => ({
+      context: {
+        referrer,
+        page,
+      },
+      environmentType,
+      network,
+      participateInMetaMetrics,
+      metaMetricsId,
+      metaMetricsSendCount,
+    }))
+  }, [network, environmentType, participateInMetaMetrics, currentPath, confirmTransactionOrigin, metaMetricsId, metaMetricsSendCount])
+
   const metricsEvent = useCallback((config = {}, overrides = {}) => {
     const { eventOpts = {} } = config
-    const { name = '' } = eventOpts
-    const { currentPath: overrideCurrentPath = '' } = overrides
-    const isSendFlow = Boolean(name.match(/^send|^confirm/u) || overrideCurrentPath.match(/send|confirm/u))
-
-    if (participateInMetaMetrics || config.isOptIn) {
-      return sendMetaMetricsEvent({
-        network,
-        environmentType,
-        activeCurrency,
-        accountType,
-        confirmTransactionOrigin,
-        metaMetricsId,
-        numberOfTokens,
-        numberOfAccounts,
-        version: global.platform.getVersion(),
-        ...config,
-        previousPath,
-        currentPath,
-        excludeMetaMetricsId: isSendFlow && !sendCountIsTrackable(metaMetricsSendCount + 1),
-        ...overrides,
-      })
+
+    const additionalContext = {}
+
+    if (overrides.currentPath) {
```

Comment:
This is no longer needed - now that you've removed the old page tracking.

Reply:
lol, didn't even pass additionalContext anywhere so i guess its a good thing it wasn't necessary